monkestation/code/modules/machining/machining_exclusives.dm

```diff
@@ -0,0 +1,229 @@
+// The makeshift pulse gun uses the backpack style. So its better here then in the pulse catagory, since backpack is more cooler about it.
+/obj/item/pulsepack
+	name = "Backpack pulse generator"
+	desc = "A pack that converts extreme amounts of power into pre pulse fusion products. Unable to be minaturized due to a lack of technology."
+	icon = 'icons/obj/weapons/guns/minigun.dmi'
+	worn_icon = 'icons/mob/clothing/back/backpack.dmi'
+	worn_icon_state = "pulsepack"
+	icon_state = "holsteredp"
+	lefthand_file = 'icons/mob/inhands/equipment/backpack_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/backpack_righthand.dmi'
+	inhand_icon_state = "pepperbackpacksec" //close enough
+	slot_flags = ITEM_SLOT_BACK
+	w_class = WEIGHT_CLASS_HUGE
+	var/obj/item/gun/energy/pulse/makeshift/gun
+	var/obj/item/stock_parts/cell/pulsepack/battery
+	var/armed = FALSE //whether the gun is attached, FALSE is attached, TRUE is the gun is wielded.
+	var/overheat = 0
+	var/overheat_max = 1
+	var/heat_diffusion = 0.13
+
+
+/obj/item/pulsepack/Initialize(mapload)
+	. = ..()
+	gun = new(src)
+	battery = new(src)
+	START_PROCESSING(SSobj, src)
+
+/obj/item/pulsepack/Destroy()
+	if(!QDELETED(gun))
+		qdel(gun)
+	gun = null
+	QDEL_NULL(battery)
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/item/pulsepack/process(seconds_per_tick)
+	overheat = max(0, overheat - heat_diffusion * seconds_per_tick)
+
+//ATTACK HAND IGNORING PARENT RETURN VALUE
+/obj/item/pulsepack/attack_hand(mob/living/carbon/user, list/modifiers)
+	if(src.loc == user)
```

Comment:
`src.var` is almost never used. in BYOND `src` is always implied.
```suggestion
	if(loc != user)
		return
```
Also use early returns. You've managed to reach 5 indents here.